### PR TITLE
refactor: All typed classes use methods with `_` prefix

### LINKED ISSUE #121: Cannot have a property on a request called validate
Get a runtime failure because when we try to call the validate function on the Concerto object we are actually accessing the field.

### PR DESCRIPTION
**Pull request by mttrbrts**
# Issue #121
In order to fix the issue where all methods in the `Typed` class and its subclasses (such as `.validate`) are reserved model properties, this PR renames all of the relevant methods to include a `_` prefix.
Fixes #121 

### Changes
- This is a *BREAKING CHANGE* and is therefore targetted for the 1.0 release.
- Changes to the API are documented in this diff, https://github.com/accordproject/concerto/compare/release-1.0...mttrbrts:mr-fix-121?expand=1#diff-46b73b28ea461a5927c3cbfe28a21bac
- I've also changed the parser to prevent identifiers starting with `_`, this guarantees that we won't have any collisions in the future.

### Flags
- @jeromesimeon and I discussed several options, listed #121, and we concluded that this change would introduce the fewest breaking changes to API clients while keeping the natural object modification behaviour for user-defined model properties, e.g. `myConcept.name = "Foo"`
- This approach also means that we don't have to maintain an extensive list of reserved property names and the difficulty of new methods introducing a breaking change in the future
- The only reserved property is `toString` which is retained to maintain compatibility with the built-in JS function.

### CODE REVIEW

File: packages/concerto-core/api.txt [hunk 1]
```diff
@@ -136,44 +136,44 @@ class TransactionDeclaration extends ClassDeclaration {
...
-   + Relationship fromURI(ModelManager,String,String,String) 
+   + boolean _isRelationship() 
+   + Relationship _fromURI(ModelManager,String,String,String) 
 }
 class Resource extends Identifiable {
    + String toString() 
-   + boolean isResource() 
-   + Object toJSON() 
+   + boolean _isResource() 
+   + Object _toJSON()
```
Comment (dselman) [inline packages/concerto-core/api.txt]: Intentional? JS calls `toJSON` automatically when you call JSON.stringify.

File: packages/concerto-core/lib/model/typed.js [hunk 2]
```diff
@@ -188,7 +188,7 @@ class Typed {
      * without using the Serializer.
      * @private
      */
-    toJSON() {
+    _toJSON() {
         throw new Error('Use Serializer.toJSON to convert resource instances to JSON objects.');
```
Comment (dselman) [inline packages/concerto-core/lib/model/typed.js]: This code will now no longer get called if you try to call JSON.stringify.

File: packages/concerto-core/api.txt (hunk 1 above)
Comment (mttrbrts) [reply packages/concerto-core/api.txt]: Thanks. No this wasn't deliberate

File: lerna.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "packages": [
     "packages/*"
   ],
-  "version": "0.82.7",
+  "version": "1.0.0-alpha.1",
```
Comment (jeromesimeon) [inline lerna.json]: Please undo, we can do an alpha release using scripts and GitHub tagging.

File: package.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
   },
   "name": "concerto",
   "description": "You must install [Lerna](https://lernajs.io) to build this multi-package repository.",
-  "version": "0.82.7",
+  "version": "1.0.0-alpha.1",
```
Comment (jeromesimeon) [inline package.json]: Please undo, we can do an alpha release using scripts and GitHub tagging.

File: packages/concerto-core/test/data/parser/classdeclaration.badidentifierprefix.cto [hunk 5]
```diff
@@ -11,16 +11,11 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-'use strict';
 
-/*eslint-disable no-unused-vars*/
```
Comment (jeromesimeon) [inline packages/concerto-core/test/data/parser/classdeclaration.badidentifierprefix.cto]: That's a weird change. Git hiccup? Maybe it's just a wrongly identified file rename.

File: packages/concerto-core/test/data/parser/functiondeclaration.bracesmissing.js [hunk 6]
```diff
@@ -1,26 +0,0 @@
-/*
```
Comment (jeromesimeon) [inline packages/concerto-core/test/data/parser/functiondeclaration.bracesmissing.js]: Are those files dead code?

File: lerna.json (hunk 3 above)
Comment (mttrbrts) [reply lerna.json]: I'm happy to back it out however, please note that:
1. This change was done to satisfy the API diff validation. If I don't incremement the version, the changelog.txt gives misleading information. i.e. breaking changes are recorded against the wrong version
2. This change was done using the scripts

File: packages/concerto-core/test/data/parser/classdeclaration.badidentifierprefix.cto (hunk 5 above)
Comment (mttrbrts) [reply packages/concerto-core/test/data/parser/classdeclaration.badidentifierprefix.cto]: Yes, this is weird. This wasn't a rename, I deleted `functiondeclaration.throwsexception.js` and created a new file `classdeclataion.badidentifierprefix.cto`, they just happen to have lines in common.

File: packages/concerto-core/test/data/parser/classdeclaration.badidentifierprefix.cto [hunk 7]
```diff
@@ -11,16 +11,11 @@
...
- * @throws {Error}
- * @exception {Error}
- */
-function onTestTransactionLulz() {
-    console.log('onTestTransaction');
+namespace com.testing
+
+asset Asset identified by id {
+  o String id
+  o String validate // This is ok
```
Comment (mttrbrts) [inline packages/concerto-core/test/data/parser/classdeclaration.badidentifierprefix.cto]: @jeromesimeon This file is used in unit tests to prove that previously reserved values can now be used.

File: packages/concerto-core/test/data/parser/functiondeclaration.bracesmissing.js (hunk 6 above)
Comment (mttrbrts) [reply packages/concerto-core/test/data/parser/functiondeclaration.bracesmissing.js]: Yes, these are legacy from when Composer supported "transaction processor functions". This makes me wonder whether there is other dead code anywhere, maybe even in the parser?

**Pull request closed without merging**